### PR TITLE
Us30 admin merchants top 5 merchants by revenue

### PR DESCRIPTION
**Pull request by amspears007**
Created model `top_five` method
Moved test data from merchant_top_five spec to rails helper
Created feature test for admin merch index page
Updated view to reflect top five customers and their total revenue.

### CODE REVIEW

**Review by abwmn — APPROVED** [review]
looks great! wow, what a time we had solving this user story. good job!

**Comment by abwmn** [comment]
great perseverance and resilience, team!! 🥇 🦾

File: app/models/merchant.rb [hunk 1]
```diff
@@ -54,6 +54,15 @@ def uniq_invoices
     invoices.distinct
   end
 
+  def self.top_five
```
Comment (thawley2) [inline app/models/merchant.rb:57]: Awesome, glad you all got this figured out!!!

File: app/controllers/admin/merchants_controller.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class Admin::MerchantsController < ApplicationController
   def index
     @merchants = Merchant.all
+    @top_five = @merchants.top_five
```
Comment (thawley2) [inline app/controllers/admin/merchants_controller.rb:4]: Just to keep with convention and could be a possible refactor later.  This method (top_five) could be called on @merchants in the view to keep the controller 'looking fit and skinny'.  
Curious what you all think about it.

**Review by thawley2 — APPROVED** [review]
Nice job you two !!

File: app/models/merchant.rb [hunk 3]
```diff
@@ -54,6 +54,15 @@ def uniq_invoices
...
+        joins(invoices: :transactions)
+        .where('transactions.result = ?', 'success')
+        .select("merchants.id, merchants.name, SUM(invoice_items.unit_price * invoice_items.quantity) as tot_revenue")
```
Comment (sicknacco) [inline app/models/merchant.rb:60]: Killed it! You got the `as` to work!

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -95,6 +95,29 @@ def merchant2_test_data
...
+
+  create_invoice_items_and_transactions(@merchant1, 1000)
+  create_invoice_items_and_transactions(@merchant2, 2000)
+  create_invoice_items_and_transactions(@merchant3, 3000)
+  create_invoice_items_and_transactions(@merchant4, 4000)
+  create_invoice_items_and_transactions(@merchant5, 5000)
+  create_invoice_items_and_transactions(@merchant6, 6000)
+end
+
+def create_invoice_items_and_transactions(merchant, revenue)
```
Comment (sicknacco) [inline spec/rails_helper.rb:114]: This is cool!

**Review by sicknacco — APPROVED** [review]

**Pull request merged**